partitura/utils/music.py

```diff
@@ -471,6 +193,88 @@ def _transpose_note(note, interval):
         )
 
 
+def transpose_note_old(step, alter, interval):
+    """
+    Transpose a note by a given interval without changing the octave or creating a Note Object.
+
+
+    Parameters
+    ----------
+    step: str
+        The step of the pitch, e.g. C, D, E, etc.
+    alter: int
+        The alteration of the pitch, e.g. -2, -1, 0, 1, 2 etc.
+    interval: Interval
+        The interval to transpose by.
+
+    Returns
+    -------
+    new_step: str
+        The new step of the pitch, e.g. C, D, E, etc.
+    new_alter: int
+        The new alteration of the pitch, e.g. -2, -1, 0, 1, 2 etc.
+    """
+    if interval.quality + str(interval.number) == "P1":
+        new_step = step
+        new_alter = alter
+    else:
+        prev_step = step.capitalize()
+        new_step = _transpose_step(prev_step, interval.number, interval.direction)
+        prev_alter = alter if alter is not None else 0
+        prev_pc = MIDI_BASE_CLASS[prev_step.lower()] + prev_alter
+        tmp_pc = MIDI_BASE_CLASS[new_step.lower()]
+        if interval.direction == "up":
+            diff_sm = tmp_pc - prev_pc if tmp_pc >= prev_pc else tmp_pc + 12 - prev_pc
+        else:
+            diff_sm = prev_pc - tmp_pc if prev_pc >= tmp_pc else prev_pc + 12 - tmp_pc
+        new_alter = (
+            INTERVAL_TO_SEMITONES[interval.quality + str(interval.number)] - diff_sm
+        )
+    return new_step, new_alter
+
+
+def transpose_note(step, alter, interval):
+    """
+    Transpose a note by a given interval without changing the octave or creating a Note Object.
```

Comment:
I would write "without considering the octave".
And the second part "without creating a Note object is a bit weird. Maybe write in a new line "This function don't create a Note object, but just returns step and alterations attributes"